File: Fortran/experimental/translate.f90
program translate
  include 'shmem.fh'

  integer :: shmem_my_pe, shmem_n_pes
  integer :: me, npes

  integer, parameter :: target_pe = 1

  integer*8, save :: symvar
  integer*8       :: remote_symvar
  integer*8       :: shmemx_lookup_remote_addr

  call shmem_init ()
  me = shmem_my_pe ()
  npes = shmem_n_pes ()

  write (*, *) 'PE ', me, 'symvar @ ', loc(symvar)
  call sleep(1)

  if (me .eq. 0) then
     remote_symvar = shmemx_lookup_remote_addr (symvar, target_pe);
     
     write (*, *) remote_symvar
  end if

end program translate

File: c/experimental/translate.c
#include <stdio.h>
#include <unistd.h>

#include <shmem.h>
#include <shmemx.h>

static const int target_pe = 1;

int
main()
{
    int me;
    long *symvar;

    shmem_init ();
    me = shmem_my_pe ();

    symvar = shmem_malloc (sizeof (*symvar) * 4);

    printf("PE %d: symvar @ %p\n", me, symvar);
    fflush (stdout);
    sleep (1);

    if (me == 0) {
        long *remote_symvar = shmemx_lookup_remote_addr (symvar, target_pe);

        if (symvar == remote_symvar) {
            printf ("symvar @ same address on PE %d\n", target_pe);
        }
        else {
            printf("symvar @ %p is at %p on PE %d\n",
                   symvar, remote_symvar, target_pe);
        }
    }

    shmem_barrier_all ();
    shmem_free (symvar);

    return 0;
}
